bfit/fit.py

```diff
@@ -126,18 +165,24 @@ def goodness_of_fit(self, coeffs, expons):
         kullback_leibler : float
             Kullback-Leibler divergence between density and approximate model density.
             This is defined to be :math:`KL(f, g) = \int f(x) \log\bigg(\frac{f(x)}{g(x)}\bigg) dx`.
+            If :math:`g` is negative, then this returns infinity.
 
         """
         # evaluate approximate model density
         approx = self.model.evaluate(coeffs, expons)
+        if np.any(approx < 0.0):
+            # Return infinity, if the model density is negative.
+            kl = np.inf
```

Comment:
@Ali-Tehrani shouldn't this check/raise happen in the `model.evaluate` method? Because this is not the case for squared difference model, so better let fit be agnostic.

Reply:
Okay, I changed it so that KL Divergence in measure.py returns infinity if model is negative and I added two measure attributes to `_BaseFit` that is only used when computing the measures, so that redundancies are removed.